lib/queue.js

```diff
@@ -63,16 +63,19 @@ var Queue = function Queue(name, redisPort, redisHost, redisOptions){
     redisOptions.db = redisOpts.DB;
   }
 
-  if (_.isString(redisPort)) {
+  if (!_.isObject(redisPort) && _.isString(redisPort)) {
```

Comment:
In my previous comment I mean an else if, the reason is that the first if could actually change redisPort into a string for example, and then you will enter this path which would be wrong.